### DISCUSSION TITLE
Youtube violation

### DISCUSSION THREAD

**Discussion by Damigrace1** [Tyrrrz/YoutubeExplode#713] [Q&A]
Hi devs. This package looks great but i have a question about it.
I went through youtube terms and it looks like this package violates youtubes terms of service as it permits downloading and distributing of youtube content. Is this true?

Because i currently have a product in play store that leverages on this library .

**Comment by Tyrrrz** [comment]
It's legally a gray area, but I would probably expect Google to remove your app from their store if you use this, yeah.

**Comment by Damigrace1** [comment]
ok but do you think simply streaming and not downloading is legal? I am
planing to remove the download functionality.

**Reply by Tyrrrz** [reply]
Neither is probably illegal in the sense that you may get sued, but Google may still remove your app. I doubt there is a difference between streaming and downloading. The best way to find out is just to test. This is not legal advice.

**Comment by Damigrace1** [comment]
Thanks for the tip!

**Comment by gcadmes** [comment]
Besides the two of us, I wonder if there are any others who have tried to submit their apps' YouTube video downloadability. 
I'm about to submit mine to the App and Play store. I've searched for other apps, and there are literally none that download YouTube Videos.

My app is interestingly different, in that... I'm restricting specific URLs to designated YouTube Channels where I have written consent from the content creators. The app audience will be incredibly small, sure, but the app is intended for my daughter's volleyball club. We livestream each tournament to YouTube, and afterwards, the players need access to their videos so they can edit/crop their clips for college recruitment.

Only thing I'm unsure of, is if I need consent from each family member who's daughter is under the age of 18.

Thoughts?

**Reply by Tyrrrz** [reply]
I'd suggest uploading the videos to another (open) platform and streaming from there. If you have the authors' consent, you can download their videos from YouTube and re-upload elsewhere.
